src/pip/_internal/index.py

```diff
@@ -865,20 +1005,33 @@ def create(
             allow_all_prereleases=selection_prefs.allow_all_prereleases,
         )
 
+        link_collector = LinkCollector(
+            session=session,
+            search_scope=search_scope,
+        )
+
         return cls(
             candidate_prefs=candidate_prefs,
-            search_scope=search_scope,
-            session=session,
+            link_collector=link_collector,
             target_python=target_python,
             allow_yanked=selection_prefs.allow_yanked,
             format_control=selection_prefs.format_control,
             ignore_requires_python=selection_prefs.ignore_requires_python,
         )
 
+    @property
+    def search_scope(self):
+        # type: () -> SearchScope
+        return self._link_collector.search_scope
+
+    def set_search_scope(self, search_scope):
+        # type: (SearchScope) -> None
+        self._link_collector.search_scope = search_scope
```

Comment:
Would it make sense to use a property setter (`@search_scope.setter`) instead? A `set_` function implies to me the call would trigger multiple things, but in reality it is only sets an (proxied) attribute. A property setter also reads better IMO.

Reply:
It doesn't make a difference to me, IMO. Happy to change to a setter.